### ISSUE TITLE
Highlighting breaks when assigning regex literals to a prototype

### ISSUE THREAD

**Issue by dcposch** [atom/language-javascript#385]
I ran into this while working on uBlock Origin code. Specifically, this line: https://github.com/gorhill/uBlock/blob/master/src/js/cosmetic-filtering.js#L950

Here's a smaller test case that reproduces the problem:

```js
var working = /^\[(?:a|b)="[^"]+"\]$/;

function a() {}
a.prototype.broken = /^\[(?:a|b)="[^"]+"\]$/;

/* below here everything is broken */
```

Here's how that looks in Atom:
<img width="439" alt="screen shot 2016-07-09 at 9 37 19 pm" src="https://cloud.githubusercontent.com/assets/169280/16711691/5c6d9d70-461d-11e6-86eb-af86d188a301.png">

So highlighting the same regex literal works when assigned to a var, but fails when assigned to a property on a prototype.

**Comment by dcposch** [comment]
Atom 1.8.0 running on OSX 10.11.5 (El Capitan)

**Comment by dcposch** [comment]
language-javascript 0.110.0

**Comment by MaximSokolov** [comment]
This is fixed by #312. The fix will be in Atom v1.9.0

**Comment by 50Wliu** [comment]
Thanks @MaximSokolov :).
